src/actions/MapActions.js

```diff
@@ -139,3 +287,50 @@ export const clearSearchMarker = () => {
         dispatch({ type: 'CLEAR_SEARCH_MARKER' });
     }
 }
+
+/**
+ * Make a POST request to the given API endpoint. Set the map saving and error state according to
+ * what happens with the request.
+ * 
+ * @param {string} endpoint the API endpoint, starting '/api/'
+ * @param {any} body the data to include in the POST request
+ * @returns {boolean} whether the save was successful
+ */
+const saveMapRequest = (endpoint, body) => {
+    return async (dispatch, getState) => {
+        const currentSaveError = getState().mapMeta.saveError;
+        dispatch({ type: 'MAP_SAVING' });
+
+        try {
+            await axios.post(`${constants.ROOT_URL}${endpoint}`, body, getAuthHeader());
+
+            dispatch({
+                type: 'MAP_SAVED',
+                payload: {
+                    timestamp: moment().format("HH:mm")
+                }
+            });
+
+            if (currentSaveError && endpoint !== '/api/user/map/save/') {
+                console.log('There was previously a save error so save the whole map');
+                dispatch(saveCurrentMap());
+            }
+
+            return true;
+        } catch (err) {
+            console.error(`There was an error in ${endpoint} POST request`, err);
+        }
+
+        dispatch({ type: 'MAP_SAVE_ERROR' });
+        return false;
+    }
+}
+
+const shortenTimestamp = (timestamp) => {
+    const isToday = moment(timestamp).isSame(moment(), 'day');
+    if (isToday) {
+        return moment(timestamp).format("HH:mm");
+    } else {
+        return moment(timestamp).format("DD/MM/YY");
+    }
```

Comment:
do you want to close or add a comment in #157 now that there's more stuff in the actions and presumably moved from components?

Reply:
Added a bit more detail to #157